src/connector/src/sink/remote.rs

```diff
@@ -770,34 +771,31 @@ impl EmbeddedConnectorClient {
 
         let jvm = self.jvm;
         std::thread::spawn(move || {
-            let mut env = match jvm
-                .attach_current_thread()
-                .context("failed to attach current thread")
-            {
-                Ok(env) => env,
-                Err(e) => {
-                    let _ = response_tx.blocking_send(Err(e));
-                    return;
-                }
-            };
+            let result = execute_with_jni_env(jvm, |env| {
+                let result = call_static_method!(
+                    env,
+                    class_name,
+                    method_name,
+                    {{void}, {long requestRx, long responseTx}},
+                    &mut request_rx as *mut JniReceiverType<REQ>,
+                    &mut response_tx as *mut JniSenderType<RSP>
+                );
 
-            let result = call_static_method!(
-                env,
-                class_name,
-                method_name,
-                {{void}, {long requestRx, long responseTx}},
-                &mut request_rx as *mut JniReceiverType<REQ>,
-                &mut response_tx as *mut JniSenderType<RSP>
-            );
-
-            match result {
-                Ok(_) => {
-                    tracing::info!("end of jni call {}::{}", class_name, method_name);
-                }
-                Err(e) => {
-                    tracing::error!(error = %e.as_report(), "jni call error");
-                }
-            };
+                match result {
+                    Ok(_) => {
+                        tracing::info!("end of jni call {}::{}", class_name, method_name);
+                    }
+                    Err(e) => {
+                        tracing::error!(error = %e.as_report(), "jni call error");
+                    }
+                };
+
+                Ok(())
+            });
+
+            if let Err(e) = result {
+                let _ = response_tx.blocking_send(Err(e));
+            }
```

Comment:
@wenym1 Do I correctly handle the error in this case?

Reply:
LGTM